src/arch/all/memchr.rs

```diff
@@ -24,6 +24,7 @@ expected match frequency.
 */
 
 use crate::{arch::generic::memchr as generic, ext::Pointer};
+use core::cmp;
```

Comment:
I think you can probably use `a.min(b)` in the places where you use `cmp` here.

Otherwise, the `use core::cmp` should go above the `use crate::{...}` import with an empty line separating them. (See how other files in this repo are structured.)